src/Symfony/Component/DependencyInjection/Compiler/AutowireRequiredMethodsPass.php

```diff
@@ -50,7 +50,7 @@ protected function processValue($value, $isRoot = false)
             while (true) {
                 if (false !== $doc = $r->getDocComment()) {
                     if (false !== stripos($doc, '@required') && preg_match('#(?:^/\*\*|\n\s*+\*)\s*+@required(?:\s|\*/$)#i', $doc)) {
-                        $value->addMethodCall($reflectionMethod->name);
+                        $value->addMethodCall($reflectionMethod->name, [], preg_match('#(?:^/\*\*|\n\s*+\*)\s*+@return\s++static[\s\*]#i', $doc));
```

Comment:
We should also support the return `self` return type and PHPDoc isn't it?

Reply:
I'm not sure actually. `self` isn't compatible with child classes and means something else to me.

Reply:
Would allowing multiple documented return types be within the scope of this?
Eg, `@return static|SomeInterface`?

Reply:
`@return static` is pretty much the only agreed declaration for withers. Anything could mean something else and shouldn't be supported IMHO, if it mean anything.

Reply:
@nicolas-grekas it is compatible with child classes: https://3v4l.org/RDg8X

Reply:
Technically yes, but semantically no: eg autocompletion wouldn't suggest method on children.

Reply:
Also, we'll probably need to support more advanced use cases at some point: https://github.com/zendframework/zend-diactoros/blob/master/src/RequestTrait.php#L252

Reply:
I'm really unsure. If some code is not expressed in a clear enough way, we shouldn't try better.
There is always manual configuration for custom cases, I see zero need to play guesswork.